api/src/services/generate-xform.js

```diff
@@ -205,51 +206,39 @@ const generate = formXml => {
     .then(([ form, model ]) => ({ form, model }));
 };
 
-const updateAttachment = (doc, updated, name, type) => {
-  const attachmentData = doc._attachments &&
-                         doc._attachments[name] &&
-                         doc._attachments[name].data &&
-                         doc._attachments[name].data.toString();
-  if (attachmentData === updated) {
-    return false;
+const addGeneratedAttachments = (doc, updated, outdated) => {
+  if (!updated || (
+    updated.form.toString() === outdated.form.toString() &&
+    updated.model.toString() === outdated.model.toString())) {
+    return;
   }
-  doc._attachments[name] = {
-    data: Buffer.from(updated),
-    content_type: type
+
+  doc._attachments['form.html'] = {
+    data: Buffer.from(updated.form),
+    content_type: 'text/html'
   };
-  return true;
+  doc._attachments['model.xml'] = {
+    data: Buffer.from(updated.model),
+    content_type: 'text/xml'
+  };
+  return doc;
 };
 
-const updateAttachmentsIfRequired = (doc, updated) => {
-  const formUpdated = updateAttachment(doc, updated.form, 'form.html', 'text/html');
-  const modelUpdated = updateAttachment(doc, updated.model, 'model.xml', 'text/xml');
-  return formUpdated || modelUpdated;
-};
+const updateAttachments = async (doc) => {
+  const enketoForm = getEnketoForm(doc);
+  if (enketoForm) {
+    const name = formsService.getXFormAttachmentName(doc);
+    const rawXML = await db.medic.getAttachment(doc._id, name, { rev: doc._rev });
+    const form = await db.medic.getAttachment(doc._id, 'form.html', { rev: doc._rev });
```

Comment:
It's possible that any of these two calls fail with a 404 if the form has just been uploaded. This is causing a lot of e2e test failures now. 
You will need to wrap this in a try-catch and always create these attachments if they don't yet exist.